cmd/main.go

```diff
@@ -84,21 +92,34 @@ func main() {
 	if cfg.TLSCert != "" && cfg.TLSKey != "" {
 		srv.TLSConfig = nil
 	}
+
 	go func() {
-		slog.Info("starting server", "addr", cfg.Listen)
+		log.Info("starting server", "addr", cfg.Listen)
+
 		var err error
-		// TODO: find a better way to set this.
 		if cfg.Listen == ":https" {
 			err = srv.ListenAndServeTLS(cfg.TLSCert, cfg.TLSKey)
 		} else {
 			err = srv.ListenAndServe()
 		}
 		if err != nil {
 			if errors.Is(err, http.ErrServerClosed) {
-				slog.Info("server shut down")
+				log.Info("server shut down")
+				return
+			}
+			log.Error("could not start server", "err", err)
+			os.Exit(1)
+		}
+	}()
+
+	go func() {
+		err := startGRPCServer(log, cfg, grpcProxyHandler)
```

Comment:
I'm thinking we should use errgroups here instead of directly spinning go routines

Reply:
Makes total sense to me. It's done, also added a graceful shutdown on the GRPC proxy as it was missing and terminating only because the main routine terminated as well.